### ISSUE TITLE
How to disable the redundant id check named "Evaluate"? 

### ISSUE THREAD

**Issue by jml** [ndmitchell/hlint#507]
I'm trying to make a good, default `.hlint.yaml` for projects that use [Protolude](https://github.com/sdiehl/protolude) as their prelude (see https://github.com/sdiehl/protolude/issues/89).

Protolude exports `id :: a -> a` as `identity :: a -> a`, leaving `id` to be used for business logic (c.f. the [Protolude FAQ](https://github.com/sdiehl/protolude#faqs)). 

This means that rules like the following are not relevant:

```yaml
    - warn: {lhs: id x, rhs: x, side: not (isTypeApp x), name: Evaluate}
    - warn: {lhs: id . x, rhs: x, name: Redundant id}
    - warn: {lhs: x . id, rhs: x, name: Redundant id}
```

Disabling the latter two is straightforward, `- ignore: {name: Redundant id}` does the trick. I am struggling to disable the first.

```yaml
    - ignore: {lhs: id x, rhs: x, side: not (isTypeApp x), name: Evaluate}
```

doesn't appear to work: `hlint` still reports an `id businessObj => businessObj` warning.

A blanket ban of all "Evaluate" rules (via `- ignore: {name: Evaluate}`) works, but then I need to copy  the 25 other evaluate rules into my project's `.hlint.yaml` and give them a different name (e.g. `Evaluate'`).

I had a quick poke around in the source but couldn't determine which was the relevant bit.

My questions are:

- is there an easier way of doing what I want? (replacing the rules for redundant id with redundant identity).
- if not, would you be amenable to a PR that provides an easier way?
- if so, could you please give me some guidance on that PR? (which bits of the code to look at, essential properties of any solution)

Thanks!

PS. hlint is a great piece of software. It has been improving my Haskell code and my Haskell skill for years. Thank you.

**Comment by ndmitchell** [comment]
The name of the hint could equally be Redundant id, and the move from id to identity seems common in alternative preludes - I've pushed a patch to rename it.
